homeassistant/components/solarlog/entity.py

```diff
@@ -0,0 +1,71 @@
+"""Entities for SolarLog integration."""
+
+from __future__ import annotations
+
+from homeassistant.components.sensor import SensorEntityDescription
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import slugify
+
+from .const import DOMAIN
+from .coordinator import SolarLogCoordinator
+
+
+class SolarLogCoordinatorEntity(CoordinatorEntity[SolarLogCoordinator]):
```

Comment:
Should we split this class in a SolarLogcontrollerentity and a base entity? This way you don't have to duplicate the _attr_has_entity_name thing (and it looks a bit nicer)